.github/workflows/integration_tests.yml

```diff
@@ -21,60 +25,96 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v5
         with:
-          go-version: 'stable'
-      - run: go version
-      - run: make deps
+          go-version: '1.x'
+
+      - name: Install Dependencies
+        run: |
+          make deps
+          sudo apt-get update -y
+          sudo apt-get install -y build-essential
 
-      - name: Update system packages
-        run: sudo apt-get update -y
+      - name: Install go-junit-report
+        run: go install github.com/jstemmer/go-junit-report/v2@latest
+
+      # Note: test tags are not in alphabetical order, it was divided to optimize test execution time
+      - name: Set Test Scope for each Account
+        run: |
+          case "${{ matrix.user }}" in 
+            "USER_1")
+              echo "TEST_TAGS=acceptance,backup,domain,domainrecord,domains,domainzonefile,firewall,firewalldevice,firewalls,helper,instance" >> $GITHUB_ENV
+              echo "LINODE_TOKEN=${{ secrets.LINODE_TOKEN_USER_1 }}" >> $GITHUB_ENV
+              ;;
+            "USER_2")
+              echo "TEST_TAGS=instancenetworking,instancesharedips,instancetype,instancetypes,ipv6range,ipv6ranges,kernel,kernels,lke" >> $GITHUB_ENV
+              echo "LINODE_TOKEN=${{ secrets.LINODE_TOKEN_USER_2 }}" >> $GITHUB_ENV
+              ;;
+            "USER_3")
+              echo "TEST_TAGS=instanceconfig,instancedisk,instanceip,nb,nbconfig,nbconfigs,nbnode,nbs,networkingip,obj,objbucket,objcluster,objkey,profile,rdns" >> $GITHUB_ENV
+              echo "LINODE_TOKEN=${{ secrets.LINODE_TOKEN_USER_3 }}" >> $GITHUB_ENV
+              ;;
+            "USER_4")
+              echo "TEST_TAGS=image,images,lkeclusters,lkenodepool,lkeversions,region,regions,sshkey,sshkeys,stackscript,stackscripts,token,user,users,vlan,volume,volumes,vpc,vpcs" >> $GITHUB_ENV
+              echo "LINODE_TOKEN=${{ secrets.LINODE_TOKEN_USER_4 }}" >> $GITHUB_ENV
+              ;;
+          esac
```

Comment:
Curious about if it can be placed in a script file

Reply:
I was thinking something similar but didn't want to risk exposing any of the secrets and writing this in the Makefile or shell script was a bit messy. I will keep this mind though and implement a cleaner solution if possible in future